zio-http/src/main/scala/zio/http/netty/NettyBodyWriter.scala

```diff
@@ -0,0 +1,72 @@
+package zio.http.netty
+
+import io.netty.buffer.Unpooled
+import io.netty.channel._
+import io.netty.handler.codec.http.LastHttpContent
+import zio._
+import zio.http.Body
+import zio.http.Body._
+
+object NettyBodyWriter {
+
+  def write(body: Body, ctx: ChannelHandlerContext): Task[Boolean] = body match {
+    case body: ByteBufBody            =>
+      ZIO.attempt(ctx.write(body.byteBufData): Unit).as(false)
+    case body: FileBody               =>
+      ZIO.attempt {
+        val file = body.fileData
+        // Write the content.
+        ctx.write(new DefaultFileRegion(file, 0, file.length()))
+
+        // Write the end marker.
+        ctx.writeAndFlush(LastHttpContent.EMPTY_LAST_CONTENT): Unit
+        true
+      }
+    case AsyncBody(async)             =>
+      ZIO
+        .attempt(async { (ctx, msg, isLast) =>
+          ctx.writeAndFlush(msg)
+          if (isLast) ctx.writeAndFlush(LastHttpContent.EMPTY_LAST_CONTENT): Unit
+          else ctx.read(): Unit
+        })
+        .as(true)
+    case AsciiStringBody(asciiString) =>
+      ZIO.attempt(ctx.write(Unpooled.wrappedBuffer(asciiString.array())): Unit).as(false)
+    case StreamBody(stream)           =>
+      for {
+        _ <- stream.runForeachChunk(c => ZIO.succeed(ctx.writeAndFlush(Unpooled.wrappedBuffer(c.toArray))))
+        _ <- ZIO.attempt(ctx.writeAndFlush(LastHttpContent.EMPTY_LAST_CONTENT))
+      } yield true
```

Comment:
`for` comprehensions yield a final `map` that can be eliminated. In performance-sensitive code, it's best to avoid them. Here you can fuse the `true` into the previous `ZIO.attempt`, and also eliminate the `ZIO.succeed` in favor of `ZIO.succeedNow`.

Reply:
None of the operations on `ChannelHandlerContext` will ever throw an exception, instead it will notify the underlying `ChannelFuture`.  So, in the interest of performance I will look at converting all calls to the context operations wrapped with either `attempt` or `succeed` to be `succeedNow`.